### PR TITLE
[Android] Bump Gradle version to 7.6.3

### PR DESCRIPTION
**Pull request by camsim99**
Part of fix for https://github.com/flutter/flutter/issues/138336 by bumping the Gradle version of all package example apps (plus `image_picker_android` plugin) to 7.6.3 from those that had a version below that to fix security vulnerability.

Also fixes a bug I found while using the `update-dependencies` packages tool command that caused it to not catch all of the `gradle-wrapper.properties` files when running with the `gradle` dependency + added a test for the fix.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by reidbaker — APPROVED** [review]

**Comment by camsim99** [comment]
I just realized most of these packages do not need version changes, so I need to undo those changes then get this landed.

**Review by gspencergoog — APPROVED** [review]

File: packages/animations/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Updates example app Gradle version to 7.6.3.
```
Comment (stuartmorgan) [inline packages/animations/CHANGELOG.md:3]: This is uninteresting to clients; we can revert all these CHANGELOGs.

File: packages/image_picker/image_picker_android/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.8.9+1
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker_android/CHANGELOG.md:1]: Except this one :)

**Review by stuartmorgan — APPROVED** [review]
LGTM with changelogs removed

**Comment by stuartmorgan** [comment]
changelog override: Minor example app configuration changes that wouldn't need to be copied by clients aren't necessary to highlight.

**Review by tarrinneal — APPROVED** [review]
number get bigger!

**Review by gmackall — APPROVED** [review]

**Review by Piinks — APPROVED** [review]
LGTM!

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**